### PR TITLE
#1285 - include more default roles

### PR DESCRIPTION
**Pull request by stpaultim**
Adding an editor role to the standard install profile

### CODE REVIEW

**Comment by stpaultim** [comment]
I got another PR mixed up with this one, trying to sort that out now.

File: core/profiles/standard/standard.install [hunk 1]
```diff
@@ -346,14 +346,17 @@ function standard_install() {
 
   // Enable default permissions for system roles.
   $filtered_html_permission = filter_permission_name($filtered_html_format);
-  user_role_grant_permissions(BACKDROP_ANONYMOUS_ROLE, array('access content', 'access comments', $filtered_html_permission));
-  user_role_grant_permissions(BACKDROP_AUTHENTICATED_ROLE, array('access content', 'access comments', 'post comments', 'skip comment approval', 'upload editor images', $filtered_html_permission));
+  user_role_grant_permissions(BACKDROP_ANONYMOUS_ROLE, array('access content',
```
Comment (klonos) [inline core/profiles/standard/standard.install]: We only break inline comment lines at 80 chars. If you want to shorten this, the use "helper" array variables, like `$anonymous_permissions = array('access content', 'access comments', $filtered_html_permission);` and `$authenticated_permissions = array(bla bla bla);`. You can then break these arrays like so:

```php
$anonymous_permissions = array(
  'access content', 
  'access comments', 
  $filtered_html_permission,
);

...

$authenticated_permissions = array(
  'access content', 
  'access comments', 
  'post comments',
  'skip comment approval', 
  'upload editor images',
  $filtered_html_permission,
);
```


...and then do this:

```suggestion
  user_role_grant_permissions(BACKDROP_ANONYMOUS_ROLE, $anon_permissions);

  ...
  user_role_grant_permissions(BACKDROP_AUTHENTICATED_ROLE, $authenticated_permissions);
```

File: core/profiles/standard/standard.install [hunk 2]
```diff
@@ -364,6 +367,23 @@ function standard_install() {
...
 
+  // Create a default editor role for users who need to be able to edit content but do not need to access
+  // advanced admin features.
+  $admin_role = new stdClass();
+  $admin_role->name = 'editor';
+  $admin_role->label = st('Editor');
+  $admin_role->weight = 2;
+  user_role_save($admin_role);
+
+  user_role_grant_permissions("editor", array('view own unpublished content',
```
Comment (klonos) [inline core/profiles/standard/standard.install]: ...same as previous comment:

```php
  $editor_permissions = array(
    'view own unpublished content', 
    'view any unpublished content',
    'view revisions',
    'revert revisions', 
    'delete revisions',
    'create page content',
    'edit own page content', 
    'edit any page content',
    'delete own page content', 
    'delete any page content',
    'create post content',
    'edit own post content', 
    'edit any post content',
    'delete own post content', 
    'delete any post content',
    'access dashboard', 
    'view the administration theme',
    'access content overview',
  );
```

...then do:
```suggestion
  user_role_grant_permissions('editor', $editor_permissions);
```

File: core/profiles/standard/standard.install [hunk 3]
```diff
@@ -364,6 +367,23 @@ function standard_install() {
     ->fields(array('uid' => 1, 'role' => $admin_role->name))
     ->execute();
 
+  // Create a default editor role for users who need to be able to edit content but do not need to access
+  // advanced admin features.
+  $admin_role = new stdClass();
```
Comment (klonos) [inline core/profiles/standard/standard.install]: All these variables should be `$editor_role` instead:

```suggestion
  $editor_role = new stdClass();
```

File: core/profiles/standard/standard.install [hunk 4]
```diff
@@ -364,6 +367,23 @@ function standard_install() {
...
+  $admin_role->name = 'editor';
+  $admin_role->label = st('Editor');
+  $admin_role->weight = 2;
+  user_role_save($admin_role);
```
Comment (klonos) [inline core/profiles/standard/standard.install]:
```suggestion
  user_role_save($editor_role);
```

File: core/profiles/standard/standard.install (hunk 3 above)
Comment (stpaultim) [reply core/profiles/standard/standard.install]: I did that and it didn't work. It broke the site. Will try again to be sure.

Comment (stpaultim) [reply core/profiles/standard/standard.install]: I am trying again to see if it was user error the first time.

Comment (stpaultim) [reply core/profiles/standard/standard.install]: This time it's working. I must have done something wrong the first time.

File: core/profiles/standard/standard.install [hunk 5]
```diff
@@ -346,14 +346,27 @@ function standard_install() {
 
   // Enable default permissions for system roles.
   $filtered_html_permission = filter_permission_name($filtered_html_format);
-  user_role_grant_permissions(BACKDROP_ANONYMOUS_ROLE, array('access content', 'access comments', $filtered_html_permission));
-  user_role_grant_permissions(BACKDROP_AUTHENTICATED_ROLE, array('access content', 'access comments', 'post comments', 'skip comment approval', 'upload editor images', $filtered_html_permission));
+  $anonymous_permissions = array(
+    'access content',
+    'access comments',
+    $filtered_html_permission
```
Comment (klonos) [inline core/profiles/standard/standard.install:352]: Coding standards dictate that we add a comma after the last entry in arrays, when they are in this format (see https://api.backdropcms.org/php-standards#arrays):

```suggestion
    $filtered_html_permission,
```

File: core/profiles/standard/standard.install [hunk 6]
```diff
@@ -346,14 +346,27 @@ function standard_install() {
...
+  );
+  $authenticated_permissions = array(
+    'access content',
+    'access comments',
+    'post comments',
+    'skip comment approval',
+    'upload editor images',
+    $filtered_html_permission
```
Comment (klonos) [inline core/profiles/standard/standard.install:360]: ...same as previous comment:

```suggestion
    $filtered_html_permission,
```

File: core/profiles/standard/standard.install [hunk 7]
```diff
@@ -364,6 +377,45 @@ function standard_install() {
...
+    'access administration bar',
+    'access content overview',
+    'access administration pages',
+    'access site in maintenance mode',
+    'flush caches',
+    'edit terms in tags',
+    'delete terms in tags',
+    'administer taxonomy',
+    'search content',
+    'use advanced search'
```
Comment (klonos) [inline core/profiles/standard/standard.install:415]:
```suggestion
    'use advanced search',
```

**Comment by quicksketch** [comment]
I addressed @klonos's feedback and added tests over in https://github.com/backdrop/backdrop/pull/2837.

**Pull request closed without merging**